gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -803,7 +807,11 @@ public EnumTypeAdapter(final Class<T> classOfT) {
         in.nextNull();
         return null;
       }
-      return nameToConstant.get(in.nextString());
+      String key = in.nextString();
+      T constant = nameToConstant.get(key);
+      if (constant == null)
```

Comment:
Google style requires a space between `if` and `(` (which I added) and also braces around the body. Alternatively, you could do this:

```java
return (constant == null) ? stringToConstant.get(key) : constant;
```

Reply:
Sorry to quibble about this, but can you add braces around the body of the `if`, or rewrite using `?:`?

Reply:
Changed to `?:`